inst/atime/tests.R

```diff
@@ -0,0 +1,68 @@
+pkg.edit.fun = quote(function(old.Package, new.Package, sha, new.pkg.path) {
+      pkg_find_replace <- function(glob, FIND, REPLACE) {
+        atime::glob_find_replace(file.path(new.pkg.path, glob), FIND, REPLACE)
+      }
+      Package_regex <- gsub(".", "_?", old.Package, fixed = TRUE)
+      Package_ <- gsub(".", "_", old.Package, fixed = TRUE)
+      new.Package_ <- paste0(Package_, "_", sha)
+      pkg_find_replace(
+        "DESCRIPTION", 
+        paste0("Package:\\s+", old.Package),
+        paste("Package:", new.Package))
+      pkg_find_replace(
+        file.path("src", "Makevars.*in"),
+        Package_regex,
+        new.Package_)
+      pkg_find_replace(
+        file.path("R", "onLoad.R"),
+        Package_regex,
+        new.Package_)
+      pkg_find_replace(
+        file.path("R", "onLoad.R"),
+        sprintf('packageVersion\\("%s"\\)', old.Package),
+        sprintf('packageVersion\\("%s"\\)', new.Package))
+      pkg_find_replace(
+        file.path("src", "init.c"),
+        paste0("R_init_", Package_regex),
+        paste0("R_init_", gsub("[.]", "_", new.Package_)))
+      pkg_find_replace(
+        "NAMESPACE",
+        sprintf('useDynLib\\("?%s"?', Package_regex),
+        paste0('useDynLib(', new.Package_))
+    })
+
+test.list <- list(
+  # Performance regression fixed in: https://github.com/Rdatatable/data.table/pull/4440
+  "Test regression fixed in #4440" = list(
+    pkg.edit.fun = pkg.edit.fun,
+    N = 10^seq(3,8),
+    setup = quote({
+      set.seed(1L)
+      dt <- data.table(a = sample(N, N))
+      setindex(dt, a)
+    }),
+    expr = quote(data.table:::shallow(dt)),
+    "Before"="9d3b9202fddb980345025a4f6ac451ed26a423be",
```

Comment:
for the real PR, please include comments on the same line as each commit ID, which explain where the commit ID comes from, with links, for example
```r
  "slow"="d47a83fb2e25582e508f191f87a31ca81b736b57",#Parent of the first commit in the PR that fixes the issue 
  "fast"="958e3dd3cba7c259220aa653bef4beb8ad74b239") # last commit in the PR that fixes the issue (https://github.com/Rdatatable/data.table/pull/4488/commits)
```
from https://github.com/DorisAmoakohene/PerformanceTest_data.table/blob/master/Working%20Examples/PR%204488/Selecting%20from%20data.table%20by%20row%20is%20very%20slow%20%233735%20-%20Copy.Rmd

Reply:
or
```r

"Before"="19b7866112614db53eb3e909c097407d91cd6738",#parent of the regression commit (https://github.com/Rdatatable/data.table/commit/0895fa247afcf6b38044bd5f56c0d209691ddb31), which is the parent of the first commit in the PR that causes the issue (https://github.com/Rdatatable/data.table/pull/5493/commits) 
  "Regression"="0895fa247afcf6b38044bd5f56c0d209691ddb31", #The regression commit is parent of the first commit in the PR that fixes the issue  (https://github.com/Rdatatable/data.table/pull/5493/commits)
  "Fixed"="1e03fe7b890e63da9651d997ea52548c90b3ae32")# last commit of the PR that fixes the issue (https://github.com/Rdatatable/data.table/pull/5493/commits)
```
from https://github.com/DorisAmoakohene/PerformanceTest_data.table/commit/177279e12f502380d25d2ff8d3cb9dd4f19ec9da

Reply:
> or
> 
> ```r
> "Before"="19b7866112614db53eb3e909c097407d91cd6738",#parent of the regression commit (https://github.com/Rdatatable/data.table/commit/0895fa247afcf6b38044bd5f56c0d209691ddb31), which is the parent of the first commit in the PR that causes the issue (https://github.com/Rdatatable/data.table/pull/5493/commits) 
>   "Regression"="0895fa247afcf6b38044bd5f56c0d209691ddb31", #The regression commit is parent of the first commit in the PR that fixes the issue  (https://github.com/Rdatatable/data.table/pull/5493/commits)
>   "Fixed"="1e03fe7b890e63da9651d997ea52548c90b3ae32")# last commit of the PR that fixes the issue (https://github.com/Rdatatable/data.table/pull/5493/commits)
> ```
> 
> from [DorisAmoakohene/PerformanceTest_data.table@177279e](https://github.com/DorisAmoakohene/PerformanceTest_data.table/commit/177279e12f502380d25d2ff8d3cb9dd4f19ec9da)

The SHA used in 'Fixed' for that is wrong ([not the last commit of #5463 or not a commit even in that PR](https://github.com/Rdatatable/data.table/commit/1e03fe7b890e63da9651d997ea52548c90b3ae32), and CI fails when using it)

The correct one for that is 58409197426ced4714af842650b0cc3b9e2cb842